### PR TITLE
Add cpu credits

### PR DESCRIPTION
**Pull request by reyntjensw**
To fix some small issues with the t3 family regarding cpu credits, I've added this block in the code.
By default it will be set to standard but it can be changed to unlimited.

### CODE REVIEW

**Comment by reyntjensw** [comment]
@simonrondelez I did not test it, the hacaddy is relying on this module so I had the idea creating a new release only for the cpu_credits, if you don't use it, you can keep using the previous version.

**Comment by simonrondelez** [comment]
@reyntjensw thats not a good idea. i propose you implement a check like we did for `is_ebs_optimised`

**Comment by reyntjensw** [comment]
@simonrondelez you mean the 
``` count                       = "${var.ebs_enabled ? 0 : var.instance_count}" ``` thingie?

**Comment by reyntjensw** [comment]
@simonrondelez like this? I'd rather put it in an if before the credit_specification, but that does not exist :(

File: instance/main.tf [hunk 1]
```diff
@@ -36,6 +36,7 @@ resource "aws_instance" "instance" {
   }
 
   credit_specification {
+    count  = "${module.cpu_credits_available.cpu_credits_available}"
```
Comment (iuriaranda) [inline instance/main.tf]: this won't work, `count` only works on resources or data resources, but not on nested objects.

TBH I would first test if the `credit_specification` object works on all instance types, maybe it it's set to  `standard` is good enough. You can test that locally before merging.

Comment (reyntjensw) [reply instance/main.tf]: ok @iuriaranda I'll try it first

Comment (reyntjensw) [reply instance/main.tf]: @iuriaranda I've tried it and it is not complaining in my plan if I use a t2 for example

Comment (iuriaranda) [reply instance/main.tf]: and what about the apply? Don't be afraid to create an instance to test this 😛

**Comment by reyntjensw** [comment]
FYI : I did a test with a t2 and using TF plan and it did not complain

**Comment by iuriaranda** [comment]
@reyntjensw as I mentioned in [the previous comment](https://github.com/skyscrapers/terraform-instances/pull/35#discussion_r227363246) (the discussion is `outdated` so you might have missed it), what about the tf apply? Have you tried launching the instance with that setting?

**Comment by reyntjensw** [comment]
@iuriaranda as mentioned through slack this the result:

``` module.wouter.aws_instance.instance: Creating...  ami:                                       "" => "ami-47a23a30"
  associate_public_ip_address:               "" => "<computed>"
  availability_zone:                         "" => "<computed>"
  credit_specification.#:                    "" => "1"
  credit_specification.0.cpu_credits:        "" => "unlimited"
  ebs_block_device.#:                        "" => "<computed>"
  ebs_optimized:                             "" => "false"
  ephemeral_block_device.#:                  "" => "<computed>"
  get_password_data:                         "" => "false"
  iam_instance_profile:                      "" => "profile_wouter_surveyanyplace_staging"
  instance_state:                            "" => "<computed>"
  instance_type:                             "" => "t2.micro"  

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

**Pull request merged**